tests/lax_control_flow_test.py

```diff
@@ -538,6 +538,31 @@ def f(upper):
     with self.assertRaisesRegex(ValueError, "Can only use `unroll`"):
       f(10)
 
+  @parameterized.named_parameters(
+      {
+          "testcase_name": f"_{jit=}_{upper=}_{unroll=}",
+          "jit": jit,
+          "upper": upper,
+          "unroll": unroll,
+      }
+      for jit in (False, True)
+      for upper in (0, -1)
+      for unroll in (False, True, None, 1, 2)
```

Comment:
`(False, True)` is probably sufficient here.

Reply:
Not really sure about this, as I can imagine valid use cases for having integer `unroll` but sometimes having negative lengths.

The `scan` logic is tricky, too:

https://github.com/google/jax/blob/a6f26306b387ce082358e8e94fdcfd777773475f/jax/_src/lax/control_flow/loops.py#L271

But I have reduced the cases to `(False, True)` for now.